### PR TITLE
[RSDK-7790] updated micro-rdk-modular-driver-examples

### PR DESCRIPTION
**Pull request by allisonschiang**
-updated example so it can compile

### CODE REVIEW

File: src/moisture_sensor.rs [hunk 1]
```diff
@@ -1,49 +1,65 @@
...
+        },
+        status::{Status, StatusError},
     },
-    status::Status,
+    DoCommand,
 };
 
-#[derive(DoCommand)]
-pub struct MoistureSensor(Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>);
+//pub struct MoistureSensor{reader : dyn AnalogReader<u16,Error = SensorType>}
```
Comment (mattjperez) [inline src/moisture_sensor.rs]: rm since it's been replaced fully

File: src/moisture_sensor.rs [hunk 2]
```diff
@@ -1,49 +1,65 @@
...
+    registry.register_sensor("moisture_sensor", &MoistureSensor::from_config)?;
     Ok(())
+
+}
+#[derive(DoCommand)]
+pub struct MoistureSensor{
+    reader : AnalogReaderType<u16>
 }
 
+
```
Comment (mattjperez) [inline src/moisture_sensor.rs]: nit: rm this extra newline

File: src/moisture_sensor.rs [hunk 3]
```diff
@@ -1,49 +1,65 @@
...
+    pub fn from_config(_cfg: ConfigType, deps: Vec<Dependency>) -> Result<SensorType, SensorError> {
+        // // get board
+        // let board = registry::get_board_from_dependencies(deps)
+        //     .expect("failed to get board from dependencies");
+        // // get reader from the board (analog reader set in app.viam)
+        // let reader = board
+        //     .lock()
+        //     .unwrap()
+        //     .get_analog_reader_by_name("moisture".to_string())?;
+        // Ok(Arc::new(Mutex::new(Self {reader})))
```
Comment (mattjperez) [inline src/moisture_sensor.rs]: remove this since you reimplemented it

**Review by mattjperez — CHANGES_REQUESTED** [review]
Looking great! 
let's remove all the commented out code first

File: Cargo.toml [hunk 4]
```diff
@@ -5,13 +5,9 @@ edition = "2021"
...
 resolver = "2"
 
-[patch.crates-io]
-esp-idf-sys = {git = "https://github.com/viamrobotics/esp-idf-sys.git"}
-
 [dependencies]
-anyhow = {version = "1", features = ["backtrace"]}
 log = "0.4"
-micro-rdk = {git = "https://github.com/viamrobotics/micro-rdk.git", features = ["esp32"]}
+micro-rdk = {git = "https://github.com/viamrobotics/micro-rdk.git", version = "0.1.9-rc2", rev = "3aae9aa"}
```
Comment (acmorrow) [inline Cargo.toml]: rc3 is out

File: src/moisture_sensor.rs [hunk 5]
```diff
@@ -1,49 +1,65 @@
...
 };
 
-#[derive(DoCommand)]
-pub struct MoistureSensor(Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>);
+//pub struct MoistureSensor{reader : dyn AnalogReader<u16,Error = SensorType>}
 
-pub fn register_model(registry: &mut ComponentRegistry) -> anyhow::Result<(), RegistryError> {
-    registry.register_sensor("moisture", &MoistureSensor::from_config)?;
-    log::info!("moisture sensor registration ok");
+pub fn register_model(registry: &mut ComponentRegistry) -> Result<(), RegistryError> {
```
Comment (acmorrow) [inline src/moisture_sensor.rs]: In the other review, I believe you renamed this from `register_model` to `register_models`: https://github.com/viam-labs/micro-rdk-esp32-sensor-examples/pull/6/files#diff-a019dea011c6ec55007918304246cb63a3292f4670e1ef259bf65c6cd6a48672R23 for the per-type registration. Let's do it here too then.

File: src/moisture_sensor.rs [hunk 6]
```diff
@@ -1,49 +1,65 @@
...
+    registry.register_sensor("moisture_sensor", &MoistureSensor::from_config)?;
     Ok(())
+
+}
+#[derive(DoCommand)]
```
Comment (acmorrow) [inline src/moisture_sensor.rs:25]: Please add a blank line after the `}` before starting a new thing.

File: src/water_pump.rs [hunk 7]
```diff
@@ -22,18 +22,17 @@ pub struct WaterPump {
     led: Option<i32>,
 }
 
-pub fn register_model(registry: &mut ComponentRegistry) -> anyhow::Result<(), RegistryError> {
+pub fn register_model(registry: &mut ComponentRegistry) -> Result<(), RegistryError> {
```
Comment (acmorrow) [inline src/water_pump.rs]: Ditto `model` vs `models`.

File: src/water_pump.rs [hunk 8]
```diff
@@ -22,18 +22,17 @@ pub struct WaterPump {
...
     registry.register_motor("water_pump", &WaterPump::from_config)?;
     log::info!("water_pump motor registration ok");
     Ok(())
-
 }
 
 impl WaterPump {
-    pub fn from_config(cfg: ConfigType, deps: Vec<Dependency>) -> anyhow::Result<MotorType> {
+    pub fn from_config(cfg: ConfigType, deps: Vec<Dependency>) -> Result<MotorType, crate::water_pump::MotorError> {
```
Comment (acmorrow) [inline src/water_pump.rs]: Why does `crate::water_pump::MotorError` need the explicit namespacing?

**Review by acmorrow — CHANGES_REQUESTED** [review]

**Review by acmorrow — APPROVED** [review]
LGTM

**Pull request merged**